### DISCUSSION TITLE
Using Nanoid on Vercel Edge Functions fails to deploy

### DISCUSSION THREAD

**Discussion by cill-i-am** [ai/nanoid#443] [Q&A] [answered]
I'm trying to deploy a project using nanoid to VErcels Edge, but my builds are failing due to the following error:

```
Error: The Edge Function "index" is referencing unsupported modules:
--
15:20:47.949 | - nanoid: crypto
```
It looks as though Vercels edge supports the web Crypto APIs. Has anybody managed to make this work or should I call it quits and deploy to their node lambdas?

EDIT: Looks like imports from `nanoid/non-seure` work on edge

**Comment by ai** [comment, marked as answer]
Does it support `window.crypto`? Try `nanoid/index.browser.js`

**Reply by cill-i-am** [reply]
That sorted it. Thanks @ai!

**Reply by ai** [reply]
We are planning to fix this problem in next major release.

**Comment by edek37** [comment]
docker pull fthomas/scala-steward
